.NET/buildtools/checkSpec.ps1

```diff
@@ -0,0 +1,129 @@
+# Check for empty and duplicate inputs in Specs
+Param ([string]$rootFolder = (get-item $(Get-Location).Path).parent.FullName)
+
+$SpecsFile = Join-Path -Path $rootFolder -ChildPath "Specs"
+$filePattern = "*.json"
+
+$global:totalEmpty = 0
+$global:totalDuplicate = 0
+$global:emptyFileList = New-Object System.Collections.Generic.List[string]
+$global:duplicateFileList = New-Object System.Collections.Hashtable
+
+function SpecInfo()
+{
+    foreach ($file in $input) 
+    {   
+        $parentName = $file.FullName | Split-Path -parent | Split-Path -leaf
+        $typeFolder = $file.FullName | Split-Path -parent | Split-Path -parent | Split-Path -leaf
+		$ret = Get-Content $file.FullName -encoding utf8 | ConvertFrom-Json
+        CheckSpec -spec $ret -type $typeFolder -parent $parentName -name $file.Name
+    }
+
+    Write-Host("Total empty input:`t" + $global:totalEmpty)
+    Write-Host("Total duplicate input:`t" + $global:totalDuplicate)
+
+    if ($global:totalEmpty)
+    {
+        Write-Host("Contain empty input `"`" in these files")
+        foreach ($case in $global:emptyFileList)
+        {
+            Write-Host($case)
+        }
+    }
+
+    if ($global:totalDuplicate)
+    {
+        Write-Host("Contain duplicate input in these files")
+        foreach ($key in $global:duplicateFileList.keys)
+        {
+            Write-Host("$($key)")
+            foreach ($case in $global:duplicateFileList[$key])
+            {
+                Write-Host($case)
+            }
+        }
+    }
+
+    if ($global:totalEmpty -OR $global:totalDuplicate)
+    {
+        if ($global:totalEmpty) { exit 1 }
+        elseif ($global:totalDuplicate) { exit 2}
+        else { exit 3} 
+    }
+}
+
+function CheckSpec($spec, $type, $parent, $name)
+{
+    $emptyNum = 0
+    $duplicateNum = 0
+    $inputDict = New-Object System.Collections.Hashtable
+    $duplicateList = New-Object System.Collections.Generic.List[string]
+    $isDateTime = $false
+    $fileName = "$($type)/$($parent)/$($name)"
+
+    if ($type -eq "DateTime")
+    {
+        $isDateTime = $true
+    }
+
+    foreach ($line in $spec)
```

Comment:
$line could be renamed to $testCase as it's not really a line from the text file and actually a full object for the whole case.